vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java

```diff
@@ -1771,19 +1812,23 @@ private void insertWorkflowDefinition(
       ca.on.oicr.gsi.vidarr.api.ProvenanceWorkflowRun<ExternalMultiVersionKey> run) {
     return DSL.using(configuration)
         .insertInto(WORKFLOW_RUN)
-        .set(WORKFLOW_RUN.HASH_ID, run.getId())
-        .set(WORKFLOW_RUN.WORKFLOW_VERSION_ID, workflowId)
-        .set(WORKFLOW_RUN.ENGINE_PARAMETERS, run.getEngineParameters())
+        .set(WORKFLOW_RUN.HASH_ID, param("hashId", run.getId()))
+        .set(WORKFLOW_RUN.WORKFLOW_VERSION_ID, param("workflowVersionId", workflowId))
+        .set(
+            WORKFLOW_RUN.ENGINE_PARAMETERS,
+            run.getEngineParameters()) // JsonNode and ObjectNode are not accepted by param
```

Comment:
Can you use a Mapper to dump the engine parameters to a String for use in param()?

Reply:
The column code generated by jooq for public.workflow_run.engine_parameters has the expected type as JsonNode so it complains when it's a string.